src/libutil-c/nix_api_util_internal.h

```diff
@@ -10,6 +10,7 @@
 struct nix_c_context
 {
     nix_err last_err_code = NIX_OK;
+    /** The last error message. Always check last_err_code. This may not have been cleared, so that clearing is fast. */
     std::optional<std::string> last_err = {};
```

Comment:
idea: Could make a getter method (not for external world, just for internal usage) that performs that check first.